app/helpers/helpers.rb

```diff
@@ -0,0 +1,9 @@
+class Helpers
+  def self.current_user(session_hash) # uses the user_id from the session hash to find the user in the database and return that user.
+    @user = User.find_by_id(session_hash[:user_id])
+  end
+
+  def self.is_logged_in?(session_hash)  # returns true if the user_id is in the session hash and false if not.
+    !!session_hash[:user_id]
```

Comment:
It's better to rely on the successful call to `current_user` than only introspecting on the `session[:id]` value. In theory, a user might have an invalid `id` in their session which would make `logged_in?` return true even though that user would not be findable later. Also, because `current_user` is memoized, it makes sense to call it as early in the request cycle as possible, which generally is the first call `logged_in?`.

```ruby
def logged_in?
  !!current_user
end
```